### PR TITLE
implement string utility

### PR DESCRIPTION
**Pull request by kungyao**

### CODE REVIEW

**Comment by younger15** [comment]
Viewed

File: src/StringUtility.cpp [hunk 1]
```diff
@@ -4,5 +4,46 @@
...
+    std::vector<std::string> split(const std::string& str, const std::string& dels) {
+        std::vector<std::string> token(0);
+        // token_compress_on means that function will reduce the empty string.
+        boost::split(token, str, boost::is_any_of(dels), boost::token_compress_on);
+        return token;
+    }
+
+    std::string sha1(const std::string& str) {
+        unsigned char output_buf[20];
+        SHA1((unsigned char*)str.c_str(), str.length(), output_buf);
```
Comment (john0312) [inline src/StringUtility.cpp:24]: API Manual recommends using the EVP_* functions.
https://linux.die.net/man/3/sha

File: src/StringUtility.cpp [hunk 2]
```diff
@@ -4,5 +4,46 @@
...
+    // https://gist.github.com/barrysteyn/7308212
+    std::string encodeBase64(const std::string& str) {
+        BIO* bmem = NULL;
+        BIO* b64 = NULL;
+        BUF_MEM* bptr = NULL;
+
+        b64 = BIO_new(BIO_f_base64());
+        //Ignore newlines
+        BIO_set_flags(b64, BIO_FLAGS_BASE64_NO_NL);
+        bmem = BIO_new(BIO_s_mem());
```
Comment (john0312) [inline src/StringUtility.cpp:38]: You might need to free this.

File: src/StringUtility.cpp (hunk 1 above)
Comment (kungyao) [reply src/StringUtility.cpp]: done

File: src/StringUtility.cpp (hunk 2 above)
Comment (kungyao) [reply src/StringUtility.cpp:46]: BIO_free_all will free entire BIO chain, so we don't need to free it specially.

**Review by younger15 — APPROVED** [review]

**Review by john0312 — APPROVED** [review]

**Pull request merged**